pyomo/solvers/plugins/solvers/direct_or_persistent_solver.py

```diff
@@ -276,3 +278,58 @@ def _get_version(self):
         if self._version is None:
             return pyomo.opt.base.solvers._extract_version('')
         return self._version
+
+
+class _IDReverseAccessIDDict(MutableMapping):
+    def __init__(self, mapping=None):
+        self._data = dict()
+        self._reversed = dict()
+
+        if mapping is not None:
+            self.update(mapping)
+
+    def __getitem__(self, key):
+        return self._data[id(key)]
+
+    def __setitem__(self, key, value):
+        self._data[id(key)] = value
+        self._reversed[id(value)] = key
+
+    def __delitem__(self, key):
+        val = self._data[id(key)]
+        del self._data[id(key)]
+        del self._reversed[id(val)]
```

Comment:
`val` is undefined in this method.

As a broader point, I am opposed to implementing a class like this for such a narrow use case (unless it is fully unit tested). Two separate ComponentMap() objects would be sufficient. I don't want to say this should hold up the release. It is just something to consider.

Reply:
@ghackebeil That is fair. I was being lazy. I'll get this updated as well.